app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -79,36 +79,51 @@ const computeType = (
 const isElementArray = ({ max, sliceName }: IElementDefinition): boolean =>
   (max && (max === "*" || +max > 1) && !sliceName) || false;
 
-const createElementNode = (
+/**
+ * Creates an ElementNode from the corresponding ElementDefinition item
+ * @param elementDefinition Element of StructureDef's snapshot used to create an ElementNode
+ * @param params Set of index and parentPath used to generate the ElementNode path attribute
+ */
+export const createElementNode = (
   elementDefinition: IElementDefinition,
-  isArrayItem?: boolean
+  params: { index?: number; parentPath?: string }
 ): ElementNode => {
+  const { index, parentPath } = params;
+  const prefixPath = parentPath || elementDefinition.path?.split(".").shift();
+  const suffixPath = elementDefinition.path?.split(".").slice(1).join(".");
+  const elementPath = `${prefixPath}${
+    prefixPath && suffixPath && "."
+  }${suffixPath}${index !== undefined ? `[${index}]` : ""}`;
   return {
     id: uuid(),
     name: elementDefinition.id?.split(".").pop() ?? "",
     children: [],
-    path: elementDefinition.path ?? "",
+    path: elementPath,
```

Comment:
why don't we include `parentPath` as well ? we won't need to pass it to `buildTree` no?

Reply:
I don't understand your question..?

Reply:
sorry my comment was too short! in `useFhirResourceTreeData` you call `buildTree`: 
```typescript
const rootNode = createElementNode(elementDefinitions[0], {
        parentPath: nodePath,
      });
      buildTree(elementDefinitions.slice(1), rootNode, rootNode, nodePath);
```
Would be it worth it to add an attribute `parentPath` in order to not have to pass `nodePath` as an argument to `buildTree` ? In the function `buildTree`, `parentPath` would be replaced by `rootNode.parentPath`

Reply:
Yep this works like a charm